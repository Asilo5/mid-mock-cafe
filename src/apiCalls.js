export const getReservations = () => {
    return fetch('http://localhost:3001/api/v1/reservations')
    .then(resp => {
        if(!resp.ok) {
            throw Error('Could not find reservations');
        }
        return resp.json();
    })
}

export const postReservation = (guestInfo) => {
  const options = {
      method: 'POST',
      body: JSON.stringify(guestInfo),
      headers: {
          'Content-Type' : 'application/json'
      }
  };

  return fetch('http://localhost:3001/api/v1/reservations', options)
  .then(resp => {
      if(!resp.ok) {
          throw Error('Could not Post reservation');
      }
      return resp.json();
  })
}