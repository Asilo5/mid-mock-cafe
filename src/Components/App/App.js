import React, { Component } from 'react';
import './App.css';
import { getReservations, postReservation, deleteReservation } from '../../apiCalls';

import Form from '../Form/Form';
import ReservationsContainer from '../ReservationsContainer/ReservationsContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
       totalReservations: [],
      //  guestInfo : {},
       error: ''
    }
  }

  componentDidMount = () => {
    getReservations()
      .then(reservations => this.setState({ totalReservations: reservations }))
      .catch(err => this.setState({ error: err.message }))
  }

  addReservations = (guest) => {
    const { totalReservations } = this.state;
    postReservation(guest)
      .then(reservation => this.setState({ totalReservations: [...totalReservations, reservation] }))
      .catch(err => this.setState({ error: err.message }))
  }

  removeReservations = (id) => {
    deleteReservation(id)
    .then(reservation => this.setState({ totalReservations: reservation }))
    .catch(err => this.setState({ error: err.message }))
  }

  render() {
    const { totalReservations } = this.state;
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form addGuestInfo={this.addGuestInfo} addReservations={this.addReservations} />
        </div>
        <div className='resy-container'>
          <ReservationsContainer totalReservations={totalReservations} removeReservations={this.removeReservations}/>
        </div>
      </div>
    )
  }
}

export default App;
