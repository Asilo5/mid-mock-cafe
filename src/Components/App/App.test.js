import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { getReservations, postReservation, deleteReservation } from '../../apiCalls';

jest.mock('../../apiCalls');

describe('App', () => {
  let mockResponse = [
     {
        id: 1,
        name: 'bob',
        date: '11/20',
        time: '11;30',
        number: 1
     }
   ]

  beforeEach(() => {
    getReservations.mockImplementation(() => {
      return Promise.resolve(mockResponse)
    })
  })

  it('should match snapshot with all information passing in correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call getReservations after component is mounted', () => {
    shallow(<App />);

    expect(getReservations).toHaveBeenCalled();
  })

  it.skip('should update state with a reservation when addReservations is called', async () => {
    const wrapper = shallow(<App />);
    
    postReservation.mockImplementation(() => {
      return Promise.resolve(mockReser )
    });

    const mockReser = { name: 'bob', date: '12/22', time: '7:30', number: 2, id: Date.now() };

    expect(wrapper.state('totalReservations')).toEqual([]);

    await wrapper.instance().addReservations(mockReser);

    expect(wrapper.state('totalReservations')).toEqual([{ name: 'bobby', date: '12/23', time: '8:30', number: 2, id: 1 }, mockReser ]);
  
  })

  it('should remove reservation from state when removeReservations is called', async () => {
    const wrapper = shallow(<App />)

    const mockReservation = [
      { name: 'bob', date: '12/22', time: '7:30', number: 2, id: 1 },
      { name: 'bobby', date: '12/12', time: '8:30', number: 3, id: 2 }
    ];

    const expected = [
      { name: 'bob', date: '12/22', time: '7:30', number: 2, id: 1 }
    ];
 
    deleteReservation.mockImplementation(() => {
      return Promise.resolve(expected)
    });

    wrapper.instance().setState({ totalReservations: mockReservation});

    await wrapper.instance().removeReservations(2);

    expect(wrapper.state('totalReservations')).toEqual(expected);

  })

})


