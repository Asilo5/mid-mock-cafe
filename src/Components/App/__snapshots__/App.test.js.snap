// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot with all information passing in correctly 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservations={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationsContainer
      removeReservations={[Function]}
      totalReservations={Array []}
    />
  </div>
</div>
`;
